### DISCUSSION TITLE
Opset version convertable issues

### DISCUSSION THREAD

**Discussion by Jusungkang** [onnx/onnx#6236] [Q&A] [answered]
I recently tried to convert my ONNX models to satisfy specific opset numbers.

The conversion code is as follows:

===================================================================
```
target_opset = 17

converted_model = version_converter.convert_version(model, target_opset)
```
===================================================================

However, I encountered an issue where the model size is too large, resulting in the following error:

===================================================================
```pytb
Traceback (most recent call last):
  File "/home/jskang/Models/Phi-3-mini-4k-instruct-onnx/cpu_and_mobile/cpu-int4-rtn-block-32-acc-level-4/test.py", line 30, in <module>
    converted_model = version_converter.convert_version(model, target_opset)
  File "/home/jskang/JSK_venv/hailo_ai_sw_suite/hailo_venv/lib/python3.10/site-packages/onnx/version_converter.py", line 172, in convert_version
    model_str = model.SerializeToString()
ValueError: Message onnx.ModelProto exceeds maximum protobuf size of 2GB: 2723080909
```
===================================================================

I tried using onnxruntime-gpu and utilizing the external data method, but neither approach worked.

Are there any solutions to resolve these issues?

Thank you for your attention.

P.S.: The reason I am focusing on opset 17 is for the supportable function for the LayerNormalization layer.

**Comment by justinchuby** [comment, marked as answer]
You can save the weights as a separate file with onnx.save with save external data = True; then load it back with load external date = False; then do version conversion with the model proto. You may refer to the api docs for more information.

**Reply by Jusungkang** [reply]
Thank you for your comments.

As you said, I found that my model was loaded with the flag option load_external_data=True.

I turned it off, and the memory issues were resolved.

However, another error occurred with the following message:

Exception has occurred: InferenceError
......
onnx.onnx_cpp2py_export.shape_inference.InferenceError: [ShapeInferenceError] (op_type
, node name: /model/constant_nodes/TensorProto.INT64/1D/1): [ShapeInferenceError] Inferred shape and existing shape differ in rank: (1) vs (0)

It seems like a shape tensor error, and I need to check the structure of the models to find where the problem lies.

But it's okay. The existing problem was solved.

Thank you for your comments.

**Reply by justinchuby** [reply]
It seems to be a known issue that may or may not be solved in the model. cc @kunal-vaishnavi
